pages/clubs/form/builder/Form.jsx

```diff
@@ -0,0 +1,397 @@
+var React = require('react');
+
+/**
+
+  A moderately complex form builer for React.
+
+  This Form component converts JSON to HTML forms, using a JSON definition
+  pass in via this.props.fields. This JSON has the following form:
+
+   {
+     fieldname1 : field definition object,
+     fieldname2 : field definition object,
+     ... : ...,
+     ...
+   }
+
+  field definition objects:
+
+   {
+     type: ["text"|"textarea"|choiceGroup"|"checkbox"|checkboxGroup"|ReactComponent],
+     label: string data,
+     placeholder: string data (optional)
+     validator: [instance or array of validator objects],
+     metered: boolean (optional),
+     controller: controller object (optional),
+   }
```

Comment:
I think there may be confusion here. I'm just talking about adding [propTypes](https://facebook.github.io/react/docs/reusable-components.html#prop-validation) to validate props passed to the new form components, not anything with the form validation itself.